packages/frontend/component/src/components/share-menu/share-menu.tsx

```diff
@@ -23,10 +25,18 @@ export interface ShareMenuProps<
 > {
   workspace: Workspace;
   currentPage: Page;
+  currentPageMode: ShareMode;
   useIsSharedPage: (
     workspaceId: string,
     pageId: string
-  ) => [isSharePage: boolean, setIsSharePage: (enable: boolean) => void];
+  ) => {
+    isSharedPage: boolean;
+    toggleShare: (enable: boolean, shareMode?: ShareMode) => void;
+  };
+  useShareMode: (
```

Comment:
Pass hook as props? so weird, maybe add comments to improve those code?

Reply:
I tried splitting it but that doesn't work, if you are not logged in the backend rejects the request and crashes the page.